### PR TITLE
Add test cases for starts_with, ends_with, contains

### PR DESCRIPTION
**Pull request by anshuldata**

### CODE REVIEW

File: dialects/datafusion.yaml [hunk 1]
```diff
@@ -236,6 +236,12 @@ scalar_functions:
   - name: right
   - name: string_split
     unsupported: True
+  - name: starts_with
+    unsupported: True
+  - name: ends_with
+    unsupported: True
```
Comment (richtia) [inline dialects/datafusion.yaml:242]: Datafusion should have starts_with/ends_with support: 
https://datafusion.apache.org/python/generated/datafusion.functions.functions.html#datafusion.functions.functions.starts_with

Comment (anshuldata) [reply dialects/datafusion.yaml]: fixed it

File: dialects/velox_presto.yaml [hunk 2]
```diff
@@ -799,6 +799,12 @@ scalar_functions:
   - name: string_split
     local_name: "split"
     unsupported: True
+  - name: starts_with
+    unsupported: True
+  - name: ends_with
+    unsupported: True
+  - name: contains
+    unsupported: True
```
Comment (richtia) [inline dialects/velox_presto.yaml:807]: Velox also supports these 3
https://facebookincubator.github.io/velox/functions.html

Could you try enabling it?

Comment (anshuldata) [reply dialects/velox_presto.yaml:807]: I tried enabling it, but hit failure as unsupported function ([here](https://github.com/voltrondata/bft/actions/runs/9890689291/job/27319466588?pr=78)).
May be python binding is missing for these ?

File: dialects/cudf.yaml [hunk 3]
```diff
@@ -220,6 +220,12 @@ scalar_functions:
     unsupported: true
   - name: string_split
     local_name: "split"
+  - name: starts_with
+    unsupported: True
+  - name: ends_with
+    unsupported: True
+  - name: contains
+    unsupported: True
```
[the remark this answers is not in the document]
Comment (anshuldata) [reply dialects/cudf.yaml]: I added for Cuda too. I don't have infra to test on cuda so haven't tested it

File: cases/string/contains.yaml [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+      - value: 'abcdefg'
+        type: string
+      - value: 'abc'
+        type: string
+    result:
+      value: true
+      type: boolean
+  - group:
+      id: basic
+      description: Basic examples as case insensitivity
```
Comment (richtia) [inline cases/string/contains.yaml:16]: After the initial definition of the group, you can just do 
```
  - group: basic
```

Comment (anshuldata) [reply cases/string/contains.yaml]: fixed it

**Review by richtia — APPROVED** [review]

**Pull request merged**